app/extensions/__init__.py

```diff
@@ -79,6 +80,50 @@
 ##########################################################################################
 
 
+class ExtraValidationSchema(Schema):
+    @validates_schema(pass_original=True)
+    def validates_schema(self, data, original_data):
+        """
+        Raise validation error if there are extra fields not defined in
+        the schema.
+
+        This is necessary because marshmallow (before 3.13.0) just
+        ignores extra fields without any validation errors.
+        """
+        if data is None:  # something wrong with the data, skip
+            return
```

Comment:
Why is this not a validation error?

Reply:
This method is called after the built-in validation is done, `data` is the cleaned data and `original_data` is the original input.  If `data` is `None`, it means the original input isn't the right type (in the test I tried `[]`) so there's nothing to validate here anymore.  There's a validation error that's raised "Invalid input type."

^ I'll add something like this in the comments